File: ha-asset/90_aws/calc_api/lambda/basic.mjs
export const handler = async (event) => {

  // 身長(cm)
  let height = parseFloat(event.queryStringParameters.height);
  // 体重(kg)
  let weight = parseFloat(event.queryStringParameters.weight);

  const HEADERS = {
    // CORS対策用
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  };

  // 妥当性チェック
  if (isNaN(height)) {
    // 必須チェック
    return {
      statusCode: 400,
      headers: HEADERS,
      body: JSON.stringify({
        result: 1,
        detail: "Invalid parameters. height is required.",
      }),
    };
  } else if (isNaN(weight)) {
    // 必須チェック
    return {
      statusCode: 400,
      headers: HEADERS,
      body: JSON.stringify({
        result: 1,
        detail: "Invalid parameters. weight is required.",
      }),
    };
  } else if (height <= 0) {
    // 数値チェック
    return {
      statusCode: 400,
      headers: HEADERS,
      body: JSON.stringify({
        result: 1,
        detail: "Invalid parameters. height is positive.",
      }),
    };
  } else if (weight <= 0) {
    // 数値チェック
    return {
      statusCode: 400,
      headers: HEADERS,
      body: JSON.stringify({
        result: 1,
        detail: "Invalid parameters. weight is positive.",
      }),
    };
  }

  // 身長(m)
  let meter_height = height / 100;
  // BMI
  let bmi = weight / (meter_height * meter_height);
  bmi = Math.floor(bmi * Math.pow(10, 3)) / Math.pow(10, 3);
  // 標準体重
  let standard_weight = 22 * (meter_height * meter_height);
  standard_weight = Math.floor(standard_weight * Math.pow(10, 3)) / Math.pow(10, 3);

  // 基礎健康情報
  let basic_health_info = {
    'height': height,
    'weight': weight,
    'bmi': bmi,
    'standard_weight': standard_weight
  }

  return {
    statusCode: 200,
    headers: HEADERS,
    body: JSON.stringify({
      result: 0,
      basic_health_info: basic_health_info,
    }),
  };
};

File: ha-asset/90_aws/calc_api/lambda/basic.py
import json
import math


HEADERS = {
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
}


def _response(status_code: int, body: dict) -> dict:
    return {
        "statusCode": status_code,
        "headers": HEADERS,
        "body": json.dumps(body, ensure_ascii=False),
    }


def _floor_3(value: float) -> float:
    return math.floor(value * 1000) / 1000


def lambda_handler(event, context):
    params = event.get("queryStringParameters") or {}

    try:
        height = float(params.get("height"))
    except (TypeError, ValueError):
        height = float("nan")

    try:
        weight = float(params.get("weight"))
    except (TypeError, ValueError):
        weight = float("nan")

    if math.isnan(height):
        return _response(
            400,
            {
                "result": 1,
                "detail": "Invalid parameters. height is required.",
            },
        )
    elif math.isnan(weight):
        return _response(
            400,
            {
                "result": 1,
                "detail": "Invalid parameters. weight is required.",
            },
        )
    elif height <= 0:
        return _response(
            400,
            {
                "result": 1,
                "detail": "Invalid parameters. height is positive.",
            },
        )
    elif weight <= 0:
        return _response(
            400,
            {
                "result": 1,
                "detail": "Invalid parameters. weight is positive.",
            },
        )

    meter_height = height / 100
    bmi = _floor_3(weight / (meter_height * meter_height))
    standard_weight = _floor_3(22 * (meter_height * meter_height))

    return _response(
        200,
        {
            "result": 0,
            "basic_health_info": {
                "height": height,
                "weight": weight,
                "bmi": bmi,
                "standard_weight": standard_weight,
            },
        },
    )
